src/main/java/com/_up/megastore/services/implementations/AuthService.java

```diff
@@ -1,21 +1,66 @@
 package com._up.megastore.services.implementations;
 
+import com._up.megastore.controllers.requests.AuthRequest;
 import com._up.megastore.controllers.requests.SignUpRequest;
+import com._up.megastore.controllers.responses.AuthResponse;
+import com._up.megastore.data.model.User;
+import com._up.megastore.security.services.JWTService;
 import com._up.megastore.services.interfaces.IAuthService;
 import com._up.megastore.services.interfaces.IUserService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Date;
 
 @Service
 public class AuthService implements IAuthService {
 
   private final IUserService userService;
+  private final AuthenticationManager authenticationManager;
+  private final JWTService jwtService;
 
-  public AuthService(IUserService userService) {
+  public AuthService(IUserService userService, AuthenticationManager authenticationManager, JWTService jwtService) {
     this.userService = userService;
+      this.authenticationManager = authenticationManager;
+      this.jwtService = jwtService;
   }
 
+  @Value("${jwt.access-token-expiration}")
+  private Long accessTokenExpiration;
+
   @Override
   public void signUp(SignUpRequest signUpRequest) {
     userService.saveUser(signUpRequest);
   }
-}
+
+  @Override
+  public AuthResponse signIn(AuthRequest authRequest) {
+    authenticateUser(authRequest.username(), authRequest.password());
+
+    User user = userService.findUserByUsernameOrThrowException(authRequest.username());
+    ifUserIsNotActivatedThrowException(user);
+    String accessToken = generateAccessToken(user);
+
+    return new AuthResponse(user.getUserId(), accessToken, user.getRole());
+  }
```

Comment:
Siguiendo el comentario anterior, aca lo unico que tendriamos que hacer es generar el JWT por lo que quedaria algo asi

```suggestion
  public AuthResponse signIn(AuthRequest authRequest) {
    User user = userService.findUserByUsernameOrThrowException(authRequest.username());
    ifUserIsNotActivatedThrowException(user);
    String accessToken = generateAccessToken(user);

    return new AuthResponse(user.getUserId(), accessToken, user.getRole());
  }
```